ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -276,6 +381,154 @@ NGraphEncapsulateOp::~NGraphEncapsulateOp() {
 // OpKernel::Compute
 //---------------------------------------------------------------------------
 void NGraphEncapsulateOp::Compute(OpKernelContext* ctx) {
+  ngraph::Event event_compute("Compute", "", "");
+
+  if (m_use_parallel_executor) {
+    NGRAPH_VLOG(1) << "NGraphEncapsulateOp::Compute: Using Pipelined Executor";
+    ComputeUsingParallelExecutor(ctx);
+  } else {
+    NGRAPH_VLOG(1) << "NGraphEncapsulateOp::Compute: Using Legacy Executor";
+    ComputeUsingLegacyExecutor(ctx);
+  }
+
+  event_compute.Stop();
+  ngraph::Event::write_trace(event_compute);
+}
+
+//---------------------------------------------------------------------------
+// ComputeUsingParallelExecutor
+//---------------------------------------------------------------------------
+void NGraphEncapsulateOp::ComputeUsingParallelExecutor(OpKernelContext* ctx) {
+  std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+
+  // TF input tensor
+  std::vector<Tensor> tf_input_tensors;
+
+  for (int i = 0; i < ctx->num_inputs(); i++) {
+    tf_input_tensors.push_back(ctx->input(i));
+  }
+
+  int step_id = ctx->step_id();
+  ngraph::Event event_compile("Compile", "", "");
+
+  // Get ngraph executable and inputs information
+  bool cache_hit;
+  OP_REQUIRES_OK(ctx, m_parallel_executor->GetNgExecutable(tf_input_tensors,
+                                                           ng_exec, cache_hit));
+
+  NGRAPH_VLOG(2) << "CACHE HIT: " << PrintBool(cache_hit) << endl;
+  NGRAPH_VLOG(2) << " Step_ID: " << step_id;
+  NGRAPH_VLOG(2)
+      << "NGraphEncapsulateOp::Compute got ngraph executable for cluster id: "
+      << m_parallel_executor->GetNgraphClusterId();
+
+  event_compile.Stop();
+  ngraph::Event::write_trace(event_compile);
+
+  // Get the pipelned tensors
+  ngraph::Event event_get_tensor("Get Tensor", "", "");
+
+  std::tuple<int, PipelinedTensorVector, PipelinedTensorVector> io_tensors;
+  OP_REQUIRES_OK(
+      ctx, m_parallel_executor->GetTensorsFromPipeline(ng_exec, io_tensors));
+
```

Comment:
do we need to check if the depth in io_tensors is not -1 before proceeding?
ie
get<0>(io_tensors)>=0

Reply:
The check it's already done by the GetTensorsFromPipeline()